streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/BufferConfigInternal.java

```diff
@@ -46,4 +58,12 @@
     public Suppressed.EagerBufferConfig emitEarlyWhenFull() {
         return new EagerBufferConfigImpl(maxRecords(), maxBytes());
     }
+
+    public boolean isLoggingEnabled() {
+        return logConfig != null;
+    }
+
+    public Map<String, String> getLogConfig() {
+        return logConfig;
```

Comment:
(As you pointed out before), the store builder javadoc says `If {@code loggingEnabled} returns false, this function will always return an empty map`, so I think you'd better do:
```suggestion
        return isLoggingEnabled() ? logConfig : Collections.emptyMap();
```